app/assets/javascripts/i18n.js

```diff
@@ -497,6 +510,63 @@
     return message.replace(/_#\$#_/g, "$");
   };
 
+  // Not part of the public API
+  I18n.splitInterpolate = function(message, matches, options) {
+    var value
+    , placeholder
+    , name
+    , regex;
+
+    var buffer = [message];
+
+    while (matches.length) {
+      placeholder = matches.shift();
+      name = placeholder.replace(this.placeholder, "$1");
+
+      if (this.isSet(options[name])) {
+        value = options[name];
+      } else if (name in options) {
+        value = this.nullPlaceholder(placeholder, message, options);
+      } else {
+        value = this.missingPlaceholder(placeholder, message, options);
+      }
+
+      buffer = buffer.reduce(function(memo, x) {
+        if (typeof x === "string") {
+          return memo.concat(this.splitReplace(x, placeholder, value));
+        } else {
+          return memo.concat([x]);
+        }
+      }.bind(this), []);
+    }
+
+    return buffer;
+  };
+
+  // Not part of the public API
+  I18n.splitReplace = function(str, delim, value) {
+    var delimRegex = delim.replace(/\{/gm, "\\{").replace(/\}/gm, "\\}");
+
+    // Wrap the delimiter in parentesis. This will split and keep the delimiter.
+    var regex = new RegExp("(" + delimRegex + ")");
+
+    return str.split(regex)
+      .filter(function(x) {
+        // Splitting will leave empty elements in between two delimiters.
+        // Remove them
+        return x !== "";
+      })
+      .map(function(x) {
```

Comment:
This doesn't check that `x` is actually from a delimiter. Would it break with something like `I18n.splitReplace("name{{name}}", "{{name}}", "John")`?